doc/make.py

```diff
@@ -369,7 +382,9 @@ def main():
         args.verbosity,
         args.warnings_are_errors,
     )
-    return getattr(builder, args.command)()
```

Comment:
`builder:`  is the instance of DocBuilder class.
`args.command:`  is for the user. If the the user types with the command `linkcheck`, then args.command would get the value 'linkcheck'.

Reply:
Good so far. What does `getattr(builder, args.command)` do in that context? And what does the `()` mean at the end?

Reply:
> What does `getattr(builder, args.command)` do in that context?

If args.command is "linkcheck", then getattr(builder, args.command)() line would call the _sphinx_build method of the builder instance with the value of argument that is  "linkcheck".


> what does the `()` mean at the end?

It will execute the function at that line. The moment user types `_sphinx_build -b linkcheck` it will show output.

Reply:
> If args.command is "linkcheck", then getattr(builder, args.command)() line would call the _sphinx_build method of the builder instance with the value of argument that is "linkcheck".

That is not correct. Try to answer what `getattr(builder, args.command)` does first (note the missing `()` at the end).

> The moment user types `_sphinx_build -b linkcheck` it will show output.

That is not correct. All user input is stored in `args` and there will be no user input later.

Reply:
> what `getattr(builder, args.command)` does first

getattr(builder, args.command) retrieves the attribute named by the value of args.command from the builder instance. If args.command is "linkcheck", it would retrieve the _sphinx_build method  with "linkcheck" command, but it would not execute the method.

Reply:
If `args.command` is "linkcheck", it would retrieve the _sphinx_build method associated with the "linkcheck" command, but it would not execute the method.

The () at the end would actually call the method and execute it. So, with the () added, getattr(builder, args.command)() would both retrieve and execute the _sphinx_build method with the value of the argument "linkcheck" ??

Reply:
> it would retrieve the _sphinx_build method associated with the "linkcheck" command

That is not correct. Could you explain what you mean by that? How is the method "associated" with the command? Maybe a small example helps

```python
class Builder:
    def _sphinx_build(self):
        print("_sphinx_build")

    def linkcheck(self):
        print("linkcheck")

builder = Builder()
method = getattr(builder, "linkcheck")
print(method)
method()
```

> but it would not execute the method ??

Correct.

Reply:
1) define a class Builder with two methods: `_sphinx_build` and `linkcheck`.

2) create an instance of the `Builder` class named `builder`.

3)  use `getattr(builder, "linkcheck")` to retrieve the `linkcheck` method from the builder instance. The variable method now holds a reference to the `linkcheck` method.
4) print the value of method. method is the  reference to the `linkcheck` method of the `builder` instance
5) call `method()` which is equivalent to calling `builder.linkcheck()`. This executes the linkcheck method, which prints "linkcheck".



> Could you explain what you mean by that? How is the method "associated" with the command?

In this example, when I set command to "html", getattr(builder, args.command) retrieves the html() method from the builder instance. Similarly, if I set command to "latex", it retrieves the latex() method.

So, the term "associated" means that I'm using these methods to represent different commands, and I'm retrieving and executing these methods based on the value of args.command . This creates a automatic system where different actions are performed based on user input.

Reply:
That is correct, good job :tada: Still, the following statement is a little fuzzy:

> If `args.command` is "linkcheck", it would retrieve the _sphinx_build method associated with the "linkcheck" command

If `args.command == "linkcheck"`, we are going to retrieve `DocBuilder().linkcheck()`. `_sphinx_build` does not play a role there.

Maybe your misunderstanding is related to https://github.com/pandas-dev/pandas/pull/54265#discussion_r1307079525?

Reply:
> Maybe your misunderstanding is related to [#54265 (comment)](https://github.com/pandas-dev/pandas/pull/54265#discussion_r1307079525)?

Yes, indeed.

Reply:
@pmeier

Reply:
@ggold7046 please follow my suggestion in https://github.com/pandas-dev/pandas/pull/54265#discussion_r1307079525.

Reply:
When I call subprocess.call(cmd), python will start a new subprocess , execute the command specified in cmd, and wait for the command to complete. Once the command is finished, subprocess.call() will return the return code of the command. The return code is usually 0 if the command ran successfully or a non-zero value if there was an error.

But here in the docbuild it's returning 1, so there is an error in the passing code ? 

I see a subprocess.run() function is also there. It says, this function returns a `CompletedProcess` object that contains information about the executed command, including the return code, stdout, and stderr.

Reply:
> But here in the docbuild it's returning 1, so there is an error in the passing code ?

Correct! Hand-waving over a lot of details, essentially `subprocess` lets you run "system commands" not just Python code. So, for the purpose of this PR, you can imagine that `cmd` is what you would actually put into your terminal and run manually.

Meaning, if you have

```python
cmd = ["_sphinx_build", "-b", "linkcheck"]
```

this is equivalent to running

```bash
$ _sphinx_build -b linkcheck
```

in your terminal. Do you see the issue now?

Reply:
> Do you see the issue now?

Not actually.

Reply:
There is no command `_sphinx_build`. Thus, the subprocess can never succeed. The command you need to run is `sphinx-build`, which you told me you have done before.

Reply:
But mroeschke suggested to change it :  https://github.com/pandas-dev/pandas/pull/54265#discussion_r1295018925
That's why I changed it to `_sphinx_build`.

Reply:
Nope, that is not what they suggested. The suggestion was to use the `DocBuilder._sphinx_build` method rather than repeating the the full `cmd` in `DocBuilder.linkcheck`. This is what `DocBuilder.hmtl` and `DocBuilder.latex` do as well. You can imagine `DocBuilder._sphinx_build` as a wrapper around calling `sphinx-build` from the command line, with some configuration specific to `pandas`.

Reply:
Again Doc Build failed .

Reply:
Well, you removed the actual doc building. L372